lib/SIL/SILWitnessTable.cpp

```diff
@@ -155,3 +156,18 @@ void SILWitnessTable::convertToDefinition(ArrayRef<Entry> entries,
 Identifier SILWitnessTable::getIdentifier() const {
   return Mod.getASTContext().getIdentifier(Name);
 }
+
+bool SILWitnessTable::conformanceIsSerialized(ProtocolConformance *conformance,
+                                              ResilienceStrategy strategy,
+                                              bool silSerializeWitnessTables) {
+  auto *nominal = conformance->getType()->getAnyNominal();
+  // Only serialize if the witness table is sufficiently static, andresilience
+  // is explicitly enabled for this compilation or if we serialize all eligible
+  // witness tables.
+  return (strategy == ResilienceStrategy::Resilient ||
```

Comment:
This would be more readable if it was split up into multiple if statements, like

```
if (condA)
  return true;

if (condB)
  return true;

...
```